motor/core.py

```diff
@@ -1434,10 +1434,11 @@ def _to_list(self, length, the_list, future, get_more_result):
             if reached_length or not self.alive:
                 future.set_result(the_list)
             else:
+                next_length = length - len(the_list)
```

Comment:
This isn't quite right since it will subtract *too much* from the length on the 3rd, 4th, etc run. Instead of updating the length here I think it would be cleaner to change the line above like this:
```python
            if length is None:
                n = result
            else:
-                n = min(length, result)
+                n = min(length-len(the_list), result)
```

Reply:
Ah, yes, I overlooked this way of batch modification